From e1320409e5c630b6a57b0594947db035057b7924 Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Wed, 22 Mar 2017 17:33:08 +0000
Subject: [PATCH] MORAY-409 Moray client needs to pass timeout to server for
 sql() Reviewed by: David Pacheco <dap@joyent.com> Approved by: David Pacheco
 <dap@joyent.com>

---
 lib/meta.js  | 8 ++++++--
 package.json | 2 +-
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/lib/meta.js b/lib/meta.js
index 70cc641..acae4bb 100644
--- a/lib/meta.js
+++ b/lib/meta.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 /*
@@ -142,8 +142,12 @@ function sql(rpcctx, statement, values, options) {
     assert.string(statement, 'statement');
     assert.ok(Array.isArray(values));
     assert.object(options, 'options');
+    assert.optionalNumber(options.timeout, 'options.timeout');
 
-    opts = { req_id: options.req_id || libuuid.create() };
+    opts = {
+        req_id: options.req_id || libuuid.create(),
+        timeout: options.timeout
+    };
     log = rpc.childLogger(rpcctx, opts);
     res = new events.EventEmitter();
 
diff --git a/package.json b/package.json
index 3110bf2..5887cdc 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "moray",
     "description": "Moray client library",
-    "version": "3.2.0",
+    "version": "3.2.1",
     "author": "Joyent (joyent.com)",
     "keywords": [ "moray" ],
     "main": "./lib/index.js",
-- 
2.21.0

